src/Service/OAuth2/OAuth2Service.php

```diff
@@ -247,7 +247,7 @@ public function getAccessTokenByClientCredentials(array $scope = [], array $audi
                     'client_secret' => $this->oAuth2Client->getClientSecret(),
                     'scope' => implode(' ', $scope),
                     'audience' => implode(' ', $audience),
-                    'redirect_uri' => $this->getOAuth2Client()->getRedirectUri(),
+                    //'redirect_uri' => $this->getOAuth2Client()->getRedirectUri(),
```

Comment:
Maybe not include it if it's perfectly useless: 
see: https://www.ory.sh/docs/troubleshooting/mandatory-redirect-uri

Reply:
Yes : https://www.ory.sh/docs/oauth2-oidc/client-credentials that flow is just a simple request